activerecord/lib/active_record/associations/builder/has_many.rb

```diff
@@ -7,14 +7,15 @@ def self.macro
     end
 
     def self.valid_options(options)
-      valid = super + [:counter_cache, :join_table, :index_errors]
+      valid = super + [:counter_cache, :join_table, :index_errors, :ensuring_owner_was]
       valid += [:as, :foreign_type] if options[:as]
       valid += [:through, :source, :source_type] if options[:through]
+      valid += [:ensuring_owner_was] if options[:dependent] == :destroy_async
```

Comment:
isn't it a duplicate with the addition 3 lines above?

Reply:
Yes, good catch @vendethiel !  Feel free to open a PR ❤️